### PR TITLE
Introduce migration for Neo4j-Migrations.

### PR DESCRIPTION
**Pull request by michael-simons**
Hey there.

I would like to request an onboarding for a CLI called Neo4j-Migrations: A database refactoring tool for Neo4j, the graph database with the same name.

### CODE REVIEW

File: src/main/scala/io/sdkman/changelogs/Neo4jMigrationsMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package io.sdkman.changelogs
+
+import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.mongodb.client.MongoDatabase
+
+@ChangeLog(order = "072")
+class Neo4jMigrationsMigrations {
+
+  val candidate = "neo4j-migrations"
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/Neo4jMigrationsMigrations.scala:9]: We have an informal convention that there are no special characters in the candidate identifier. I'd suggest simply going for `neo4jmigrations`.

**Review by marc0der — COMMENTED** [review]
Hi @michael-simons! I'm happy to help you with the onboarding process. I'm not sure if you have seen it, but we have a [wiki page](https://github.com/sdkman/sdkman-cli/wiki/Vendor-onboarding-process) that describes the easiest way to do that. Please shout if anything is unclear and I'd be happy to help.

File: src/main/scala/io/sdkman/changelogs/Neo4jMigrationsMigrations.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Candidate(
+      candidate = candidate,
+      name = "Neo4j-Migrations",
+      description =
+        "Neo4j-Migrations is a database migration and refactoring tool that allows running Cypher scripts and programmatic refactorings in a controlled and repeatable fashion against one or more Neo4j database.",
+      websiteUrl = "https://neo4j.com/labs/neo4j-migrations/"
+    ).insert()
+
+  @ChangeSet(order = "002", id = "002__add_1_5_0", author = "michael-simons")
+  def migration002(implicit db: MongoDatabase) = {
```
[the remark this answers is not in the document]
Comment (michael-simons) [reply src/main/scala/io/sdkman/changelogs/Neo4jMigrationsMigrations.scala:26]: Ok, kewl. 
I followed the wiki and the readme, and had a look at a couple of other migrations, but I wasn't sure if a default candidate is necessary or not. Will change.

**Review by marc0der — APPROVED** [review]
LGTM!

**Pull request merged**